modules/vault-cluster/main.tf

```diff
@@ -295,7 +299,7 @@ data "aws_iam_policy_document" "vault_s3" {
 }
 
 data "aws_iam_policy_document" "vault_auto_unseal_kms" {
-  count = var.enable_auto_unseal ? 1 : 0
+  count = var.enable_auto_unseal && var.iam_instance_profile_arn == null ? 1 : 0
```

Comment:
Same with all of these? It doesn't seem like they should be conditional upon a custom instance profile?

Reply:
https://github.com/hashicorp/terraform-aws-vault/pull/170#issuecomment-555454004